Avoid crash that happens for some users in qsb, qmake, and
potentially other Qt tools when ran under sandbox leading
to build failures for qtdeclarative and other packages.

Note that this is the initial (simple) version of the patch
to ensure no issues, a more proper fix should land upstream
eventually.

https://bugs.gentoo.org/908809
https://bugs.gentoo.org/908816
https://bugs.gentoo.org/913493
https://bugs.gentoo.org/915695
https://codereview.qt-project.org/c/qt/qtbase/+/513140
From: Huang Rui <vowstar@users.noreply.github.com>
Date: Fri, 20 Oct 2023 10:16:01 +0800
Subject: [PATCH] folkfd_linux,c: set stack size to 8192
--- a/src/3rdparty/forkfd/forkfd_linux.c
+++ b/src/3rdparty/forkfd/forkfd_linux.c
@@ -157,7 +157,7 @@ static int system_forkfd_pidfd_set_flags(int pidfd, int flags)
 
 int system_vforkfd(int flags, pid_t *ppid, int (*childFn)(void *), void *token, int *system)
 {
-    __attribute__((aligned(64))) char childStack[4096];
+    __attribute__((aligned(64))) char childStack[8192];
     pid_t pid;
     int pidfd;
     unsigned long cloneflags = CLONE_PIDFD | CLONE_VFORK | CLONE_VM | SIGCHLD;
